src/Control/Middleware/TrustedProxyMiddleware.php

```diff
@@ -36,102 +60,126 @@ public function getTrustedProxyIPs()
      */
     public function setTrustedProxyIPs($trustedProxyIPs)
     {
         $this->trustedProxyIPs = $trustedProxyIPs;
+        return $this;
     }
 
     /**
-     * Return the comma-separated list of headers from which to lookup the hostname
+     * Return the array of headers from which to lookup the hostname
      *
-     * @return string
+     * @return array
      */
     public function getProxyHostHeaders()
     {
         return $this->proxyHostHeaders;
     }
 
     /**
-     * Set the comma-separated list of headers from which to lookup the hostname
+     * Set the array of headers from which to lookup the hostname
+     * Can also specify comma-separated list as a single string.
      *
-     * @param $proxyHostHeaders string
+     * @param array|string $proxyHostHeaders
+     * @return $this
      */
     public function setProxyHostHeaders($proxyHostHeaders)
     {
-        $this->proxyHostHeaders = $proxyHostHeaders;
+        if (is_string($proxyHostHeaders)) {
+            $proxyHostHeaders = preg_split('/ *, */', $proxyHostHeaders);
+        }
+        $this->proxyHostHeaders = $proxyHostHeaders ?: [];
+        return $this;
     }
 
     /**
-     * Return the comma-separated list of headers from which to lookup the client IP
+     * Return the array of headers from which to lookup the client IP
      *
-     * @return string
+     * @return array
      */
     public function getProxyIPHeaders()
     {
         return $this->proxyIPHeaders;
     }
 
     /**
-     * Set the comma-separated list of headers from which to lookup the client IP
+     * Set the array of headers from which to lookup the client IP
+     * Can also specify comma-separated list as a single string.
      *
-     * @param $proxyIPHeaders string
+     * @param array|string $proxyIPHeaders
+     * @return $this
      */
     public function setProxyIPHeaders($proxyIPHeaders)
     {
-        $this->proxyIPHeaders = $proxyIPHeaders;
+        if (is_string($proxyIPHeaders)) {
+            $proxyIPHeaders = preg_split('/ *, */', $proxyIPHeaders);
+        }
+        $this->proxyIPHeaders = $proxyIPHeaders ?: [];
+        return $this;
     }
 
     /**
-     * Return the comma-separated list of headers from which to lookup the client scheme (http/https)
+     * Return the array of headers from which to lookup the client scheme (http/https)
      *
-     * @return string
+     * @return array
      */
     public function getProxySchemeHeaders()
     {
         return $this->proxySchemeHeaders;
     }
 
     /**
-     * Set the comma-separated list of headers from which to lookup the client scheme (http/https)
+     * Set array of headers from which to lookup the client scheme (http/https)
+     * Can also specify comma-separated list as a single string.
      *
-     * @param $proxySchemeHeaders string
+     * @param array|string $proxySchemeHeaders
+     * @return $this
      */
     public function setProxySchemeHeaders($proxySchemeHeaders)
```

Comment:
Oh, I was trying to consolidate the discrepancy between the documented args and the actual type. We can fix it to array only if you like